app/models/remote_theme.rb

```diff
@@ -452,8 +457,16 @@ def update_theme_color_schemes(theme, schemes)
     end
 
     if missing_scheme_names.length > 0
-      ColorScheme.where(id: missing_scheme_names.values).delete_all
-      # we may have stuff pointed at the incorrect scheme?
+      to_be_deleted_ids = []
+      missing_scheme_names.values.each do |cs|
+        if (base = existing_schemes.find { |s| s.id == cs.base_scheme_id && s.remote_copy })
+          to_be_deleted_ids << cs.base_scheme_id
+        else
+          to_be_deleted_ids << cs.id
+        end
```

Comment:
I'm abit confused by the changes here. When a remote copy of a scheme exists, why do we only destroy the remote copy but not the original scheme?

Reply:
My thinking here is that if a remote copy exists, then that means an admin has modified a theme-owned palette and may expect it to stay around even if the theme no longer ships that particular palette, so in that case, which just destroy the copy (which is invisible from the UI), detach the modified palette from the theme and keep it around.